scripts/train_ner.py

```diff
@@ -0,0 +1,131 @@
+import sys
+import random
+import os
+
+import plac
+import tqdm
+import spacy
+from spacy.gold import minibatch
+from pathlib import Path
+
+sys.path.insert(0, os.path.dirname(os.path.abspath(os.path.join(__file__, os.pardir))))
+
+from SciSpaCy.data_util import read_med_mentions
+from SciSpaCy.per_class_scorer import PerClassScorer
+
+@plac.annotations(
+        output_dir=("output directory to store model in", "positional", None, str),
+        train_path=("location of training data", "positional", None, str),
+        dev_path=("location of development data", "option", None, str),
+        test_path=("location of test data", "option", "test", str),
+        model=("location of base model", "option", "model", str),
+        n_iter=("number of iterations", "option", "n", int),
+        batch_size=("batch size", "option", "b", int))
+def main(output_dir: str,
+         train_path: str,
+         dev_path: str=None,
+         test_path: str=None,
+         model: str=None,
+         n_iter: int=100,
+         batch_size: int=32):
+
+    os.makedirs(output_dir, exist_ok=True)
+    if test_path is not None:
+        test_data = read_med_mentions(test_path)
+        test(model, test_data)
+    else:
+        train_data = read_med_mentions(train_path)
+        if dev_path is None:
+            train_examples = len(train_data)
+            split = int(train_examples * 0.2)
+            dev_data = train_data[:split]
+            train_data = train_data[split:]
+        else:
+            dev_data = read_med_mentions(dev_path)
+
+        train(model, train_data, dev_data, output_dir, batch_size, n_iter)
+
+def test(model, test_data):
+    nlp = spacy.load(model)
+    print("Loaded model '%s'" % model)
+    evaluate(nlp, test_data)
+
+def train(model, train_data, dev_data, output_dir, batch_size, n_iter):
+    """Load the model, set up the pipeline and train the entity recognizer."""
+    if model is not None:
+        nlp = spacy.load(model)  # load existing spaCy model
+        print("Loaded model '%s'" % model)
+    else:
+        nlp = spacy.blank('en')  # create blank Language class
+        print("Created blank 'en' model")
```

Comment:
you may need to load a specific vocab for this to work with other components

Reply:
For the `vocab_path` in the parser retraining script what do you pass in? I might leave this because eventually all these retraining scripts will need to take a `model_path` parameter which will have been generated from `spacy init-model` for some biomed specific vocab anyway.

Reply:
yeah, that is reasonable. I currently pass in the vocab from combined_all_model and then replace the vocab in combined_all_model with the vocab from the retrained model. there is definitely a better way to handle that, just my hacky solution for now